### PR TITLE
Add tests & update instructions.

### PR DESCRIPTION
**Pull request by jorgebucaran**
- Add tests with 100% coverage.
- Improve instructions, match style used in hyperapp docs.
- Use prettier to format code.

cc @okwolf.

### CODE REVIEW

**Comment by okwolf** [comment]
@jbucaran any reason for using Upper Case for `Logger`? It looks like it should be a component. Since it's a function, I'm used to going with lower case.

File: index.js [hunk 1]
```diff
@@ -1,65 +1,37 @@
...
-    };
-  };
+      }
+    }
+  }
+}
 
-  return logger;
-});
+function log(prevState, action, nextState) {
```
Comment (okwolf) [inline index.js]: A nit, but I like to define functions prior to using them.

**Comment by jorgebucaran** [comment]
@okwolf In hyperapp we don't use classes / constructors, so we don't generally need to follow that convention and it's also how the Router works too. 

The module exports a function so you can always...

```js
import logger from "hyperapp-router"
```

...if you want though.

**Comment by okwolf** [comment]
@jbucaran I guess it matters more for the global for use in browsers.

**Comment by jorgebucaran** [comment]
@okwolf React's global is also React and ReactDOM. It also looks good, but I guess that's a matter of taste too.

File: index.js (hunk 1 above)
Comment (jorgebucaran) [reply index.js]: Done.

**Comment by okwolf** [comment]
@jbucaran should the global for Hyperapp be `Hyperapp` then? 😉

**Comment by jorgebucaran** [comment]
@okwolf Yes, I think so. That should be the way to go. Or not, maybe only mixins?

/cc @MatejMazur @Zaceno @Jamen @Andyrj @Selfup

**Comment by okwolf** [comment]
@jbucaran is upper case used by many mixins in the wild outside of Hyperapp? I see [some](https://github.com/zaceno/hyperapp-transitions) [examples](https://github.com/andyrj/hyperapp-redux-devtools) to the [contrary](https://github.com/jamen/hyperapp-persist).

**Comment by jorgebucaran** [comment]
@okwolf Most modules are a single default export so if the exported global is lowercase or uppercase, I don't mind.

But in the router, we export two things and the default export is actually { Router, Link }. Link is a component, and those **must be** uppercase. The Router is mixin, not a component, so it doesn't really matter. I made it uppercase because I felt it was more aesthetically pleasing though, but I can change my mind if that's how people like it.

**Comment by jorgebucaran** [comment]
@okwolf Let's go with lowercase then. I'll update the router later.

**Review by okwolf — APPROVED** [review]

**Comment by okwolf** [comment]
@jbucaran thanks for the updates. I'm ready to merge if you're happy.

**Comment by jorgebucaran** [comment]
👍

**Pull request merged**